### ISSUE TITLE
SDL2.dll error

### ISSUE THREAD

**Issue by philstopford** [mellinoe/veldrid#109]
This seems similar to #102, but the answers there didn't seem to help. I've attached the project for completeness (simply trying to see how I could use veldrid inside an Eto.Forms UI).

I used the Visual Studio converter to switch from packages.config (as suggested in #102), but the attempt to run the debug binary fails in the same way - it's unable to find SDL2.

It didn't seem right to tack on to the old ticket, so I open this one instead.
[veldrid_eto_test.zip](https://github.com/mellinoe/veldrid/files/2255395/veldrid_eto_test.zip)

**Comment by mellinoe** [comment]
Hey @philstopford. If you are targeting .NET Framework, unfortunately it will only be able to locate / deploy the native binary for SDL2 if you have selected a specific architecture. By default, the project will target "Any CPU". If you go into your project settings and select "Platform target: x86", or "x64", then things should work correctly.

Let me know if that helps.

**Comment by philstopford** [comment]
That got me closer - thanks!

**Comment by mellinoe** [comment]
@philstopford Glad to hear it. Did you hit any other problems, or is this one solved?

**Comment by philstopford** [comment]
I have to implement a different way, I think, to make Eto.Forms happy. If you have an example of an integration using .NET without imgui, that would be useful reference - the docs leave this up to the user from what I can tell.

**Comment by mellinoe** [comment]
Unfortunately, I'm not familiar with Eto.Forms, so I'm not sure how it would or could work there. It's possible to embed Veldrid into WinForms / WPF / UWP because they support embedding native "windows" which makes it easy, or they directly support D3D11 embedding.

Without some kind of hook or extension point in Eto.Forms, I'm not sure how it would work.

**Comment by tzachshabtay** [comment]
From what I understand Eto is using an underlying platform (WinForms, WPF, GTK and the various Xamarin platforms), so you can maybe use the hook(s) of whatever platform(s) you're targeting and then follow the instructions here: https://github.com/picoe/Eto/wiki/Embedding-Eto.Forms-in-an-existing-application#using-native-controls-within-etoforms

**Comment by mellinoe** [comment]
I wrote this a while back as an example of how to embed into WPF: https://gist.github.com/mellinoe/2ba885476f51ef867715e5aa64b390de

It builds on top of some example code for SharpDX I found: https://github.com/Smartrak/WpfSharpDxControl/blob/master/WpfSharpDxControl/Win32HwndControl.cs

This is all very specific to D3D11. If you wanted to actually build something portable and cross-platform (which used Vulkan, OpenGL, Metal, etc. on other platforms), then it would be more difficult and you'd likely need either some low-level hook in GTK, or specific widget features in Eto.
